torch/_inductor/config.py

```diff
@@ -293,7 +293,7 @@ def bundle_triton_into_fx_graph_cache_default() -> Optional[bool]:
 inter_node_bw = 25
 
 # enable slow autotuning passes to select algorithms
-max_autotune = os.environ.get("TORCHINDUCTOR_MAX_AUTOTUNE") == "1"
+max_autotune = Config(False, env_name_default="TORCHINDUCTOR_MAX_AUTOTUNE")
```

Comment:
Why do we need this ? the existing minifier serializes config settings without this change

Reply:
@eellison 

The problem is that the minifier cannot produce the same config as the run time config.

For example, if you run the script below like

If you just run the code below directly, like `TORCHINDUCTOR_MAX_AUTOTUNE=1 python test.py`, the configs that the minifier dumps doesn't have `torch._inductor.config.max_autotune = True`, even though you have env variable `TORCHINDUCTOR_MAX_AUTOTUNE=1`.


This is because the config module takes the ` os.environ.get("TORCHINDUCTOR_MAX_AUTOTUNE") == "1"` as default and skipped this config, which is wrong. The true default value is False, and the user actually changed the value of this config with an env variable. 

```
import torch
from torch._inductor import config as inductor_config
class Model(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.fc1 = torch.nn.Linear(10, 16)
        self.relu = torch.nn.ReLU()
        self.sigmoid = torch.nn.Sigmoid()
    def forward(self, x):
        x = self.fc1(x)
        x = self.relu(x)
        x = self.sigmoid(x)
        return x
inductor_config.aot_inductor.dump_aoti_minifier = True
torch._inductor.config.triton.inject_relu_bug_TESTING_ONLY = "compile_error"
with torch.no_grad():
    model = Model().to("cuda")
    example_inputs = (torch.randn(8, 10).to("cuda"),)
```

Reply:
Most of the changes here is to add a flag when dumping the configs, so we don't skip the configs that are changed by the setting the env variables.

Reply:
I think we should be able to do the above without the `max_autotune = Config(False, env_name_default="TORCHINDUCTOR_MAX_AUTOTUNE")` changes

Reply:
> I think we should be able to do the above without the `max_autotune = Config(False, env_name_default="TORCHINDUCTOR_MAX_AUTOTUNE")` changes.. But I guess this is the standard way, so fine with me..

@eellison yeah I agree, we could do it another way, but since there is the `Config` class already, I thought we might as well just use it.

Reply:
It does work, but I would still prefer we do the general thing that covers all configs instead of just this specific one.